// apisave/index.js
var mongoose = require('mongoose');//mongoose to control mongodb
mongoose.connect('mongodb://localhost/scrapcritical');


var Report = mongoose.model('report', { title: String, link: String, description: String});



module.exports = {
	newRecord: function(elementos,callback) {//function to get the elements to store in the db
		elementos.forEach(function (element, index, array) {
			guardarDatos(element,callback)//call the function that create new records in the db
		})
	},

	getAll: function(callback) {//this function gets all the records stored in the db
		Report.find({}, function (err, docs) {
		  callback(err, docs)
		});
	},

	queryRecords: function(query,callback) {//this function helps query the records in the db
		var newQuery = new RegExp(query);
		console.log(newQuery)
		Report.find({$or: [{"title": newQuery},{"description": newQuery} ]}, function(err, docs) {
		  callback(err,docs)
		});
	}
}

var guardarDatos = function(element,callback) {
	var newReport = new Report({ title: element.title, link: element.link, description: element.description });
	newReport.save(function (err) {
	  if (err) {
	  	callback(err)
	    console.log(err);
	  } else {
	  	callback("ok")
	  }
	});
}
